### PR TITLE
2024 04 29 payment links and reference numbers

### PR DESCRIPTION
**Pull request by christophercameron-ixd**
# PR Checklist

- [X] Set yourself as the Assignee
- [X] Tag anyone you would like to review, or @forms-design or @forms-devs
- [X] Fill in the template below

---

# What

New documentation trello card: https://trello.com/c/IIZ0KDNx/1507-documentation-payment-links 

Write up the designs and decisions that brought about our current implementation of the payment links work… include any details about reference number decisions made

# How to review

1. Semantic: Do you agree with the changes?
2. Syntactic: Spelling, grammar, etc.
3. Is there anything that’s been missed?
4. Do we need to provide more detail about the changes or is this enough?

# Who can review

Anyone in the team, but useful for @C-Harry to double check

### CODE REVIEW

File: design/features/payment-links/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+**As a form processor**, I need to be able to reconcile who has filled a form and who has paid.  
+
+**As a department**, I need to collect the money owed for the services we provide, so we can cover costs of running an effective service (and clear our debt).  
+**As a department**, I need to know when to set up a Pay account and how to do it, so it is in place by the time we need it.  
+
+
+## Why  
+
+When users fill in a form they may often be asked for a payment for the service. This can be as a Bacs (Bankers Automated Clearing System) payment, cheque or may be followed up with a link or phone call to request payment. This can make the process take longer meaning the user doesn’t receive an answer or outcome in a timely manner. This process can also mean more work for internal staff as they may need to chase for payment or make follow up calls before they can proceed.
```
Comment (hannahkc) [inline design/features/payment-links/README.md:46]:
```suggestion
Many document-based forms also require a payment for the service. This can be as a Bacs (Bankers Automated Clearing System) payment, cheque or may be followed up with a link or phone call to request payment. This can make the process take longer meaning the user doesn’t receive an answer or outcome in a timely manner. This process can also mean more work for internal staff as they may need to chase for payment or make follow up calls before they can proceed.  
```
Just to make it clearer we're taking about a feature we needed to be able to digitise document-based forms?

**Review by hannahkc — COMMENTED** [review]
I've only had a very very quick scan through this as am sure Charlotte will have better perspective on this one! Added a couple of suggestions tho. Ace work!

File: design/features/payment-links/version-1.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+- [Designs](#designs)
+- [Preveiwing the form fillers journey](#preveiwing-the-form-fillers-journey)
+- [Research focus](#research-focus)
+
+___
+
+## What
+
+### As-is
+
```
[the remark this answers is not in the document]
Comment (christophercameron-ixd) [reply design/features/payment-links/version-1.md:26]: Do you think this needs to be added as the first point? Or do your updates to the points cover this now?

[the remark this answers is not in the document]
Comment (hannahkc) [reply design/features/payment-links/version-1.md:26]: Yeah sorry - the suggestion wasn't really as clear as I thought it would be! The suggestion as to add this as a first point as it's the main thing really? The other two points are sort of because of this. Does that make sense?

[the remark this answers is not in the document]
Comment (christophercameron-ixd) [reply design/features/payment-links/version-1.md:26]: Done

File: design/features/payment-links/version-1.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+When the form creator returns to the “Create a form” task list, if there is already a payment link added then this row appears with ‘Completed’ on the right side of the row. There is still no increase in number of tasks completed.  
+
+___
+
+## Preveiwing the form fillers journey
+
+### Confirmation screen preview 
+
+![“You still need to pay” heading inside a blue box replacing the usual green confirmaiton box. Screenshot](./screenshots-v1/1000-preview-confirmation-still-need-to-pay.png)
```
Comment (DavidBiddle) [inline design/features/payment-links/version-1.md:154]:
```suggestion
![“You still need to pay” heading inside a blue box replacing the usual green confirmation box. Screenshot](./screenshots-v1/1000-preview-confirmation-still-need-to-pay.png)  
```

**Review by DavidBiddle — DISMISSED** [review]
Looks good, raised one tiny typo comment

**Review by DavidBiddle — DISMISSED** [review]

File: design/features/payment-links/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  - [Why](#why)
+  - [Hypothesis](#hypothesis)
+___
+
+## What  
+
+This feature introduces new functionality to the form building tool to allow form creators to take a payment for their service. This way of taking payments is utilising the [payment page](https://payments.service.gov.uk/govuk-payment-pages/), or payment link, feature developed by GOV.UK Pay. This feature allows form fillers to pay for the service after submitting their form, making it quicker and easier to get the outcome they need.  
+
+
+### User stories
```
Comment (C-Harry) [inline design/features/payment-links/README.md:27]: We updated these user needs to make them a bit more specific and relevant - [as per this document](https://docs.google.com/document/d/1BBwjtd2rYa3aiFieIOutynjWXWzStJup2SzA2e9Sp4c/edit#heading=h.seyd9a5yt3bm)

File: design/features/payment-links/version-1.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+> “It could be up to several months before you’re ready to take payments. This depends on the PSP arrangements for your organisation.”  
+
+The next section, “Creating your payment link in GOV.UK Pay”, outlines key steps to make sure that the payment link works correctly with GOV.UK Forms:  
+
+> Once you have a GOV.UK Pay account you can “add a new service” and start creating your payment link.
+
+There is a heading-level 3, “Set up your payment link to use a form’s unique reference number”, which provides more detail about the reference numbers that GOV.UK Forms uses. This includes how to make sure these are correctly processed with the payment link provided by form creators:  
+
+> GOV.UK Forms adds a unique 8-character reference to each form submission.
```
Comment (C-Harry) [inline design/features/payment-links/version-1.md:107]: @christophercameron-ixd is there a way to add line breaks between these rows - to reflect how the content is presented on the screen - while still keeping this showing as inset text for presentation purposes here? 

I tried adding a line break but that ended up presenting it as lots of separate pieces of content (with a small grey inset text bar next to each line, rather than retaining one longer grey inset text bar that spans the different lines of content).

Comment (christophercameron-ixd) [reply design/features/payment-links/version-1.md:107]: Like you want it to be a blank line? 
To visually separate them, you just need to have the `>` at the start of the line without any text on it (if that makes sense). 

```
> Paragraph 1
> 
> Paragraph 2
```

Comment (C-Harry) [reply design/features/payment-links/version-1.md:107]: Yep, exactly - thanks!

**Review by C-Harry — APPROVED** [review]
Hi @christophercameron-ixd, I think this is looking great overall. Thanks for capturing everything so clearly and fully (yet concisely!).

I've just made a few suggested tweaks for style and readability (like shortening sentences etc.)

One slight worry I've had is that I've made the odd change to the text describing the screenshots (meaning the words within the square brackets, as below):

“Create a form” task-list page showing “Add a link to a payment page on GOV.UK Pay” marked as ‘completed’. Screenshot

But it's just occurred to me that this may have broken the links..? I hope that's not the case, but if it is you can just ignore those tweaks as they were only very minor and can be lived without - e.g. adding quotation marks or hyphenating compound modifers like 'task-list page'.

File: design/features/payment-links/version-1.md [hunk 6]
```diff
@@ -130,98 +134,118 @@ The page ends with a text input labelled “Enter the URL of your GOV.UK Pay pay
...
 ___
 
-## Preveiwing the form fillers journey
+## Previewing the form filler’s journey
 
 ### Confirmation screen preview 
 
 ![“You still need to pay” heading inside a blue box replacing the usual green confirmation box. Screenshot](./screenshots-v1/1000-preview-confirmation-still-need-to-pay.png)  
 
-This is an alternative confirmation screen used where a payment link has been added to a form. It shows a blue background panel with the heading “You still need to pay” followed by “Your form reference number is:” and the unique generated reference number for this forms submission.
```
Comment (christophercameron-ixd) [inline design/features/payment-links/version-1.md]: @C-Harry why was this removed? 
I feel like this context is missing (especially as a descriptive for anyone who struggles to see the images)

Comment (C-Harry) [reply design/features/payment-links/version-1.md]: This must have been an accident @christophercameron-ixd - I'm still pretty clunky with the Mac and occasionally use the Windows shortcuts (I'm on a Windows keyboard). So, in short, this must've been an error - apologies!!

Comment (C-Harry) [reply design/features/payment-links/version-1.md]: The only thing I'd have meant to change was adding an additional apostrophe to "forms submission" - so it would be "form's submission"

**Review by hannahkc — APPROVED** [review]
Approving so you can merge :)
